docs/how-to/observability-with-prometheus-stack.md

```diff
@@ -0,0 +1,37 @@
+# Configure Observability with Prometheus Stack
+
+The Prometheus Stack includes Grafana and Prometheus.
+
+Add helm repo:
+```sh
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+helm repo update
+```
+
+
+Install the Prometheus Stack with PodMonitor for scraping vLLM metrics:
+```sh
+helm install prometheus prometheus-community/kube-prometheus-stack -f - <<EOF
```

Comment:
It would be best to assume that the team/system that is managing kube-prometheus-stack is distinct from the team that is deploying KubeAI. We should not rely on prometheus helm values to scrape KubeAI pods. The idiomatic way of doing this is to template PodMonitor/ServiceMonitor resources from the component's helm chart that is to be monitored. (See ArgoCD as an example: https://github.com/argoproj/argo-helm/blob/3a329eae633ebf374d1aa2f6673b828848f9c994/charts/argo-cd/values.yaml#L860)

Reply:
What about doing both? An end to end guide and updating KubeAI helm chart so it can create crd when available.

Reply:
Note it also doesn't work if you do:
1. Install Prometheus Stack with default values
2. Install PodMonitor CRD


It won't automatically update the prometheus config correctly. I ended up having to delete both the prometheus operator and prometheus pod. Otherwise 2 won't work. So by doing the podMonitor in the helm values, we can avoid having to manually recycle prometheus pods.